code/game/objects/structures/stool_bed_chair_nest/bed.dm

```diff
@@ -633,3 +635,74 @@
 	R.add_fingerprint(user)
 	qdel(held)
 	held = null
+
+/obj/structure/roller_rack
+	name = "roller bed rack"
+	desc = "A rack for holding collapsed roller beds."
+	icon = 'icons/obj/rollerbed.dmi'
+	icon_state = "holder"
+	var/list/held
+	var/initial_beds = 4
+
+/obj/structure/roller_rack/Initialize()
+	. = ..()
+	LAZYINITLIST(held)
+	var/i
+	for(i=1, i<=initial_beds, i++)
+		var/obj/item/roller/RB = new /obj/item/roller(src)
+		LAZYADD(held, RB)
+	update_icon()
+
+/obj/structure/roller_rack/Destroy()
+	. = ..()
+	for(var/obj/O in held)
+		qdel(O)
+	LAZYCLEARLIST(held)
+
+/obj/structure/roller_rack/two
+	initial_beds = 2
+
+/obj/structure/roller_rack/three
+	initial_beds = 3
+
+/obj/structure/roller_rack/update_icon()
+	. = ..()
+	CutOverlays()
+	var/beds = 0
+	for(var/obj/item/roller/RB in held)
+		var/image/I = overlay_image(icon, "[icon_state]_bed_[RB.base_icon]")
+		I.pixel_x = (5 * beds)
+		beds++
+		AddOverlays(I)
+
+/obj/structure/roller_rack/examine(mob/user)
+	desc = "[initial(desc)] \nIt is holding [LAZYLEN(held)] beds."
+	. = ..()
+
+/obj/structure/roller_rack/attack_hand(mob/user)
+	if(!LAZYLEN(held))
+		to_chat(user, SPAN_NOTICE("The rack is empty."))
+		return
+
+	var/obj/item/roller/RB = held[LAZYLEN(held)]
+	user.put_in_hands(RB)
+	LAZYREMOVE(held, RB)
+	to_chat(user, SPAN_NOTICE("You retrieve \the [RB] from the rack."))
+	update_icon()
+
+/obj/structure/roller_rack/attackby(W, mob/user)
```

Comment:
No, use the appropriate and correct proc header

`/attackby(obj/item/attacking_item, mob/user)`